### ISSUE TITLE
[Device] Please add further characteristics for TuYa TS011F_plug_1 and TuYa TS0121_plug

### ISSUE THREAD

**Issue by JayJay-Master** [itavero/homebridge-z2m#404]
Please use the template and fill in all the data requested below!

**Device description**
TuYa TS011F_plug_1 and TuYa TS0121_plug

**Supported in Zigbee2MQTT?**
Is this device already supported by Zigbee2MQTT / zigbee-herdsman-converters?        YES
If this is NOT the case, please add support for the device there first.
Read there manual on [How to support new devices](https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html) for more information.
If this is the case, please provide the version in which support is present/added.

**Device model / Exposes information**
Provide the device model (also known as [exposes information](https://www.zigbee2mqtt.io/guide/usage/exposes.html)) for this particular device, as published to `zigbee2mqtt/bridge/devices` by Zigbee2MQTT.

1. TuYa TS011F_plug_1 
2. TuYa TS0121_plug

```json
	{
		"date_code": "",
		"definition": {
			"description": "Smart plug (with power monitoring)",
			"exposes": [
				{
					"features": [
						{
							"access": 7,
							"description": "On/off state of the switch",
							"name": "state",
							"property": "state",
							"type": "binary",
							"value_off": "OFF",
							"value_on": "ON",
							"value_toggle": "TOGGLE"
						}
					],
					"type": "switch"
				},
				{
					"access": 1,
					"description": "Instantaneous measured power",
					"name": "power",
					"property": "power",
					"type": "numeric",
					"unit": "W"
				},
				{
					"access": 1,
					"description": "Instantaneous measured electrical current",
					"name": "current",
					"property": "current",
					"type": "numeric",
					"unit": "A"
				},
				{
					"access": 1,
					"description": "Measured electrical potential value",
					"name": "voltage",
					"property": "voltage",
					"type": "numeric",
					"unit": "V"
				},
				{
					"access": 1,
					"description": "Sum of consumed energy",
					"name": "energy",
					"property": "energy",
					"type": "numeric",
					"unit": "kWh"
				},
				{
					"access": 3,
					"description": "Recover state after power outage",
					"name": "power_outage_memory",
					"property": "power_outage_memory",
					"type": "enum",
					"values": [
						"on",
						"off",
						"restore"
					]
				},
				{
					"access": 7,
					"description": "Plug LED indicator mode",
					"name": "indicator_mode",
					"property": "indicator_mode",
					"type": "enum",
					"values": [
						"off",
						"off/on",
						"on/off",
						"on"
					]
				},
				{
					"features": [
						{
							"access": 3,
							"description": "Enables/disables physical input on the device",
							"name": "state",
							"property": "child_lock",
							"type": "binary",
							"value_off": "UNLOCK",
							"value_on": "LOCK"
						}
					],
					"type": "lock"
				},
				{
					"access": 1,
					"description": "Link quality (signal strength)",
					"name": "linkquality",
					"property": "linkquality",
					"type": "numeric",
					"unit": "lqi",
					"value_max": 255,
					"value_min": 0
				}
			],
			"model": "TS011F_plug_1",
			"options": [
				{
					"access": 2,
					"description": "Calibrates the power value (percentual offset), takes into effect on next report of device.",
					"name": "power_calibration",
					"property": "power_calibration",
					"type": "numeric"
				},
				{
					"access": 2,
					"description": "Calibrates the current value (percentual offset), takes into effect on next report of device.",
					"name": "current_calibration",
					"property": "current_calibration",
					"type": "numeric"
				},
				{
					"access": 2,
					"description": "Calibrates the voltage value (percentual offset), takes into effect on next report of device.",
					"name": "voltage_calibration",
					"property": "voltage_calibration",
					"type": "numeric"
				}
			],
			"supports_ota": true,
			"vendor": "TuYa"
		},
		"endpoints": {
			"1": {
				"bindings": [
					{
						"cluster": "genOnOff",
						"target": {
							"endpoint": 1,
							"ieee_address": "0x00212effff0781ef",
							"type": "endpoint"
						}
					},
					{
						"cluster": "haElectricalMeasurement",
						"target": {
							"endpoint": 1,
							"ieee_address": "0x00212effff0781ef",
							"type": "endpoint"
						}
					},
					{
						"cluster": "seMetering",
						"target": {
							"endpoint": 1,
							"ieee_address": "0x00212effff0781ef",
							"type": "endpoint"
						}
					}
				],
				"clusters": {
					"input": [
						"genIdentify",
						"genGroups",
						"genScenes",
						"genOnOff",
						"seMetering",
						"haElectricalMeasurement",
						"manuSpecificTuya_3",
						"57344",
						"genBasic"
					],
					"output": [
						"genOta",
						"genTime"
					]
				},
				"configured_reportings": [
					{
						"attribute": "rmsCurrent",
						"cluster": "haElectricalMeasurement",
						"maximum_report_interval": 3600,
						"minimum_report_interval": 5,
						"reportable_change": 50
					},
					{
						"attribute": "currentSummDelivered",
						"cluster": "seMetering",
						"maximum_report_interval": 3600,
						"minimum_report_interval": 5,
						"reportable_change": [
							1,
							1
						]
					},
					{
						"attribute": "activePower",
						"cluster": "haElectricalMeasurement",
						"maximum_report_interval": "120",
						"minimum_report_interval": 5,
						"reportable_change": 10
					},
					{
						"attribute": "rmsVoltage",
						"cluster": "haElectricalMeasurement",
						"maximum_report_interval": "120",
						"minimum_report_interval": 5,
						"reportable_change": 5
					}
				],
				"scenes": []
			},
			"242": {
				"bindings": [],
				"clusters": {
					"input": [],
					"output": [
						"greenPower"
					]
				},
				"configured_reportings": [],
				"scenes": []
			}
		},
		"friendly_name": "Steckdose 5",
		"ieee_address": "0xa4c1389e2b8a719c",
		"interview_completed": true,
		"interviewing": false,
		"manufacturer": "_TZ3000_amdymr7l",
		"model_id": "TS011F",
		"network_address": 19218,
		"power_source": "Mains (single phase)",
		"supported": true,
		"type": "Router"
	}
```

**Missing features/functionality**
Please describe the features/functionality of this device that are currently not exposed to HomeKit via this plugin.

Please add following characteristics: current, power, voltage and energy. All characteristics are availible in Zigbee2MQTT. If these characteristics are not supportet in Apples Home app, they should be visible/availible in alternative EVE app or Home+ app.

**Suggested services and characteristics**
For each of the missing features, please mention to which HomeKit Services and Characteristics it should be mapped.
The [Homebridge Plugin Development documentation](https://developers.homebridge.io/) provides a nice overview of the standard HomeKit services and characteristics.
Note that some manufacturers also use custom services and characteristics (the Elgato Eve products for instance), which could also be mimicked if they have been reverse engineered.

Characteristics for EVE app should be current, power, voltage and energy.

**Comment by itavero** [comment]
Possible duplicate of #96

**Comment by JayJay-Master** [comment]
I found a hint to solve the problem with deconz-rest api an Phoscon software in
[https://forum.smartapfel.de/forum/thread/8390-blitzwolf-bw-shp13-steckdose-zigbee-mit-leistungsmessung/](https://forum.smartapfel.de/forum/thread/8390-blitzwolf-bw-shp13-steckdose-zigbee-mit-leistungsmessung/)
May be it is helpful to solve the problem.

**Comment by JayJay-Master** [comment]
Hello,
is there anybody who can implement this additionall characteristics in homebridge-z2m, so that they are available in EVE or HOME+ app?

**Comment by itavero** [comment]
If someone would be able to give #1155 a go on their system as well, that would be great.

(You can check out the branch, run `npm ci`, `npm build` and lastly `npm link` to do so).

**Comment by itavero** [comment]
This is now released in [v1.11.0-beta.11](https://github.com/itavero/homebridge-z2m/releases/tag/v1.11.0-beta.11) 🚀
